lib/python/src/bailo/helper/datacard.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from typing import Any
+
+from bailo.core.client import Client
+from bailo.core.enums import EntryKind, ModelVisibility
+from bailo.helper.entry import Entry
+
+
+class Datacard(Entry):
+    """Represent a datacard within Bailo.
+
+    :param client: A client object used to interact with Bailo
+    :param datacard_id: A unique ID for the datacard
+    :param name: Name of datacard
+    :param description: Description of datacard
+    :param visibility: Visibility of datacard, using ModelVisibility enum (e.g Public or Private), defaults to None
+    """
+
+    def __init__(
+        self,
+        client: Client,
+        datacard_id: str,
+        name: str,
+        description: str,
+        visibility: ModelVisibility | None = None,
+    ) -> None:
+        super().__init__(
+            client=client,
+            id=datacard_id,
+            name=name,
+            description=description,
+            kind=EntryKind.DATACARD,
+            visibility=visibility,
+        )
+
+        self.datacard_id = datacard_id
+
+    @classmethod
+    def create(
+        cls,
+        client: Client,
+        name: str,
+        description: str,
+        team_id: str,
+        visibility: ModelVisibility | None = None,
+    ) -> Datacard:
+        """Build a datacard from Bailo and upload it.
+
+        :param client: A client object used to interact with Bailo
+        :param name: Name of datacard
+        :param description: Description of datacard
+        :param team_id: A unique team ID
+        :param visibility: Visibility of datacard, using ModelVisibility enum (e.g Public or Private), defaults to None
+        :return: Datacard object
+        """
+        res = client.post_model(
+            name=name, kind=EntryKind.DATACARD, description=description, team_id=team_id, visibility=visibility
+        )
+        datacard = cls(
+            client=client,
+            datacard_id=res["model"]["id"],
+            name=name,
+            description=description,
+            visibility=visibility,
+        )
+
+        datacard._Entry__unpack(res["model"])
+
+        return datacard
+
+    @classmethod
+    def from_id(cls, client: Client, datacard_id: str) -> Datacard:
+        """Return an existing datacard from Bailo.
+
+        :param client: A client object used to interact with Bailo
+        :param datacard_id: A unique datacard ID
+        :return: A datacard object
+        """
+        res = client.get_model(model_id=datacard_id)["model"]
+        datacard = cls(
+            client=client,
+            datacard_id=datacard_id,
+            name=res["name"],
+            description=res["description"],
+        )
+        datacard._Entry__unpack(res)
+
+        datacard.get_card_latest()
+
+        return datacard
```

Comment:
Can use these as protected class methods in the parent class.

Reply:
Done